Index: node.module
===================================================================
RCS file: /var/cvs/Drupal-i18n/modules/node.module,v
retrieving revision 1.4
diff -u -r1.4 node.module
--- node.module	3 Dec 2004 15:35:03 -0000	1.4
+++ node.module	7 Dec 2004 13:45:28 -0000
@@ -990,7 +990,7 @@
   global $base_url, $locale;
 
   if (!$nodes) {
-    $nodes = db_query_range('SELECT n.nid FROM {node} n '. node_access_join_sql() .' WHERE '. node_access_where_sql() .' AND n.promote = 1 AND n.status = 1 ORDER BY n.created DESC', 0, 15);
+    $nodes = db_query_range('SELECT n.nid FROM {node} n '. node_access_join_sql() .' WHERE '. node_access_where_sql() . node_i18n_where_sql().' AND n.promote = 1 AND n.status = 1 ORDER BY n.created DESC', 0, 15);
   }
 
   while ($node = db_fetch_object($nodes)) {
@@ -1432,7 +1432,7 @@
  * Generate a listing of promoted nodes.
  */
 function node_page_default() {
-  $result = pager_query('SELECT DISTINCT(n.nid), n.sticky, n.created FROM {node} n '. node_access_join_sql() .' WHERE n.promote = 1 AND n.status = 1 AND '. node_access_where_sql() .' ORDER BY n.sticky DESC, n.created DESC', variable_get('default_nodes_main', 10));
+  $result = pager_query('SELECT DISTINCT(n.nid), n.sticky, n.created FROM {node} n '. node_access_join_sql() .' WHERE n.promote = 1 AND n.status = 1 AND '. node_access_where_sql() .' '. node_i18n_where_sql().' ORDER BY n.sticky DESC, n.created DESC', variable_get('default_nodes_main', 10));
 
   if (db_num_rows($result)) {
     drupal_set_html_head('<link rel="alternate" type="application/rss+xml" title="RSS" href="'. url('node/feed', NULL, NULL, TRUE) .'" />');
@@ -1725,4 +1725,11 @@
  * @} End of "defgroup node_access".
  */
 
-?>
+function node_i18n_where_sql(){
+  if( function_exists('i18n_node_where_sql')){
+    return i18n_node_where_sql();
+  } else {
+    return '';
+  }
+}
+?>
\ No newline at end of file
