Index: common.inc
===================================================================
RCS file: /var/cvs/Drupal-i18n/includes/common.inc,v
retrieving revision 1.3
diff -u -r1.3 common.inc
--- common.inc	30 Nov 2004 11:56:15 -0000	1.3
+++ common.inc	7 Dec 2004 13:49:03 -0000
@@ -1504,6 +1504,9 @@
 
   $path = drupal_get_path_alias($path);
 
+  if(function_exists('i18n_url_rewrite')){
+    $path = i18n_url_rewrite($path);
+  }
   if (isset($fragment)) {
     $fragment = '#'. $fragment;
   }
\ No newline at end of file
