Index: common.inc
===================================================================
RCS file: /cvs/drupal/drupal/includes/common.inc,v
retrieving revision 1.465
diff -u -r1.465 common.inc
--- common.inc	1 Aug 2005 05:14:05 -0000	1.465
+++ common.inc	17 Aug 2005 11:03:17 -0000
@@ -1540,9 +1540,15 @@
     // Apache.
     $script = (strpos($_SERVER['SERVER_SOFTWARE'], 'Apache') === false) ? 'index.php' : '';
   }
-
+  
   $path = drupal_get_path_alias($path);
 
+  // Hook url rewrite
+  foreach(module_implements('url_rewrite') as $module){
+    $function = $module.'_url_rewrite';
+    $function($path, $query, $fragment, $absolute);
+  }
+  
   if (isset($fragment)) {
     $fragment = '#'. $fragment;
   }
