--- menu.1	2005-02-04 00:31:02.676416680 +0100
+++ menu.inc	2005-02-04 00:31:19.646836784 +0100
@@ -198,12 +198,13 @@
 function menu_get_menu() {
   global $_menu;
   global $user;
+  global $locale;
 
   if (!isset($_menu['items'])) {
     // _menu_build() may indirectly call this function, so prevent infinite loops.
     $_menu['items'] = array();
 
-    $cid = 'menu:'. $user->uid;
+    $cid = 'menu:'. $user->uid. $locale;
     if ($cached = cache_get($cid)) {
       $_menu = unserialize($cached->data);
     }
