Index: bootstrap.inc
===================================================================
RCS file: /var/cvs/Drupal-i18n/includes/bootstrap.inc,v
retrieving revision 1.4
diff -u -r1.4 bootstrap.inc
--- bootstrap.inc	25 Dec 2004 20:08:10 -0000	1.4
+++ bootstrap.inc	25 Dec 2004 20:10:04 -0000
@@ -73,6 +73,9 @@
 function variable_get($name, $default) {
   global $conf;
 
+  if (function_exists('i18n_variable')) {
+     $name = i18n_variable($name);
+  }
   return isset($conf[$name]) ? $conf[$name] : $default;
 }
 
@@ -88,6 +91,9 @@
 function variable_set($name, $value) {
   global $conf;
 
+  if (function_exists('i18n_variable')) {
+    $name = i18n_variable($name);
+  }
   db_query("DELETE FROM {variable} WHERE name = '%s'", $name);
   db_query("INSERT INTO {variable} (name, value) VALUES ('%s', '%s')", $name, serialize($value));
   cache_clear_all('variables');
@@ -104,6 +110,9 @@
 function variable_del($name) {
   global $conf;
 
+  if (function_exists('i18n_variable')) {
+    $name = i18n_variable($name);
+  }
   db_query("DELETE FROM {variable} WHERE name = '%s'", $name);
   cache_clear_all('variables');
 
@@ -181,7 +190,7 @@
  * Store the current page in the cache.
  */
 function page_set_cache() {
-  global $user, $base_url;
+  global $user, $base_url, $i18n_url;;
 
   if (!$user->uid && $_SERVER['REQUEST_METHOD'] == 'GET') {
     // This will fail in some cases, see page_get_cache() for the explanation.
@@ -194,7 +203,7 @@
           $data = gzencode($data, FORCE_GZIP);
         }
       }
-      cache_set($base_url . request_uri(), $data, CACHE_TEMPORARY, drupal_get_headers());
+      cache_set($base_url . $i18n_url. request_uri(), $data, CACHE_TEMPORARY, drupal_get_headers());
     }
   }
 }
@@ -208,12 +217,12 @@
  * get cached either.
  */
 function page_get_cache() {
-  global $user, $base_url;
+  global $user, $base_url, $i18n_url;
 
   $cache = NULL;
 
   if (!$user->uid && $_SERVER['REQUEST_METHOD'] == 'GET' && count(drupal_set_message()) == 0) {
-    $cache = cache_get($base_url . request_uri());
+    $cache = cache_get($base_url . $i18n_url . request_uri());
 
     if (empty($cache)) {
       ob_start();
@@ -230,10 +239,11 @@
   if (variable_get('dev_timer', 0)) {
     timer_start();
   }
+  bootstrap_invoke_all('init');
 
   if (variable_get('cache', 0)) {
     if ($cache = page_get_cache()) {
-      bootstrap_invoke_all('init');
+
       // Set default values:
       $date = gmdate('D, d M Y H:i:s', $cache->created) .' GMT';
       $etag = '"'. md5($date) .'"';
\ No newline at end of file
